flexmeasures/api/v3_0/assets.py

```diff
@@ -252,12 +253,17 @@ def patch(self, asset_data: dict, id: int, db_asset: GenericAsset):
                 for attr_key, attr_value in v.items():
                     if current_attributes.get(attr_key) != attr_value:
                         audit_log_data.append(
-                            f"Attribute name: {attr_key}, Old value: {current_attributes.get(attr_key)}, New value: {attr_value}"
+                            # Truncate the event to 150 characters to account for the audit_log_data not exceeding 255 characters
```

Comment:
So each attribute in `attributes` gets potentially 150 characters? If two are changed, we already have potentially 300? We could also try `150 / num_updated_attributes` for a better indication how much space each will get.

Reply:
Good suggestion with Each updated field or attribute gets its own audit log by doing that we reduce code complexity i'm working on that, So modifying 5 attributes would result in 5 `audit_log` entries. i have a suggestion to reduce entry size do you think we should combine entries that are small in size for example: 
changing
```
{  "capacity_in_mw": 0.5, 
   "min_soc_in_mwh": 0.05, 
   "max_soc_in_mwh": 0.45, 
   "sensors_to_show": [{"title": "Test Title 1", "sensors": [1]}, {"title": "Test Title 2", "sensors": [3,2]}, {"title": "Test Title 3", "sensors": [2,5]}, {"title": "Test Title 4", "sensors": [1]} ,{"title": "Test Title 5", "sensors": [1]}]}
```
to 
```
{  "capacity_in_mw": 0.05, 
   "min_soc_in_mwh": 0.005, 
   "max_soc_in_mwh": 0.045, 
   "sensors_to_show": [{"title": "New Test Title 1", "sensors": [1]}, {"title": "Test Title 2", "sensors": [3,2]}, {"title": "Test Title 3", "sensors": [2,5]}, {"title": "Test Title 4", "sensors": [1]} ,{"title": "Test Title 5", "sensors": [1]}]}
```
would result in `audit_log` 2 entries instead of 5 : 

- `Updated asset 'toy-battery': 3 fields: Attr: capacity_in_mw, From: 0.5, To: 0.05; Attr: min_soc_in_mwh, From: 0.05, To: 5;Attr: max_soc_in_mwh, From: 0.45, To: 0.045` 


- `Updated asset 'toy-battery-1': 3 fields: Attr: sensors_to_show, From: [{'title': 'Test Title 1', 'sensors': [1]}, {'title': 'Test Title 22', 'sensors': [3, 2]}, {'title': 'Test Title 3', 'sensors': [2, 5]}, {'title': 'Test Title 4', 'sensors': [1] ... , 'sensors': [1]}], To: [{'title': 'Test Title 1', 'sensors': [1]}, {'title': 'Test Title 222', 'sensors': [3, 2]}, {'title': 'Test Title 3', 'sensors': [2, 5]}, {'title': 'Test Title 4', 'sensors': [1]}, {'title': 'Test Title 5', 'sensors': [1]}]`

or if you want to keep each update in it's own entry to keep cohesion that can also be done.

Reply:
Let's keep this simple and look to finish it soon. Then, the simple route (one entry per change attribute) is the way. We can then observe if & how this is useful in practice. Maybe not much.

It will get challenging when we edit larger things like `sensor_to_show` or the `flex-context`. But for nnow I need to move on. This was in principle a bug fix PR.

Reply:
okay, understood. I will push the fix now.